### PR TITLE
FileScan Analyzer

### PR DESCRIPTION
**Pull request by uzaxirr**
# Description

Please include a summary of the change.

## Related issues
Please add related issues.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [ ] The pull request is for the branch develop
- [ ] A new analyzer or connector was added, in which case:
    - [ ] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](./Advanced-Usage.md) was updated (in case the analyzer/connector provides additional optional configuration).
    - [ ] Secrets were added in [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the [Installation](./Installation.md) docs, if necessary.
    - [ ] If the analyzer/connector requires mocked testing, `_monkeypatch()` was used in it's class to apply the necessary decorators.
    - [ ] If a File analyzer was added, it's name was explicitly defined in [test_file_scripts.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_file_scripts.py) (not required for Observable Analyzers).
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [ ] The tests gave 0 errors.
- [ ] `Black` gave 0 errors.
- [ ] `Flake` gave 0 errors.
- [ ] The commits were squashed into a single one (optional, they will be squashed anyway by the maintainer)
  
### Important Rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review

# Real World Example

Please delete if the PR is for bug fixing.
Otherwise, please provide the resulting raw JSON of a finished analysis (and, if you like, a screenshot of the results). This is to allow the maintainers to understand how the analyzer works.

### CODE REVIEW

File: api_app/analyzers_manager/file_analyzers/filescan.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            if response.status_code != 200:
+                logger.info(f"Error: {response.status_code}")
+                time.sleep(5)
+                continue
+            else:
+                post_sucess = True
+                break
+        if post_sucess:
+            json_response = response.json()
+            self.task_id = json_response["flow_id"]
```
Comment (sp35) [inline api_app/analyzers_manager/file_analyzers/filescan.py]: what if task_id is never set

File: api_app/analyzers_manager/file_analyzers/filescan.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    def __filescan_poll_result(self):
+        logger.info(
+            f"polling result for ({self.filename},{self.md5}), task_id: {self.task_id}"
+        )
+        # get_sucess = False
+        for chance in range(self.max_get_tries):
+            logger.info(
+                f"polling request #{chance+1} for file ({self.filename}, {self.md5})"
+            )
+            url = self.request_url + "api/scan/" + str(self.task_id) + "report"
```
Comment (sp35) [inline api_app/analyzers_manager/file_analyzers/filescan.py]: most probably the reason you are getting a 404 is the above

Comment (uzaxirr) [reply api_app/analyzers_manager/file_analyzers/filescan.py]: Yes u are correct it should be `url = self.request_url + "api/scan/" + str(self.task_id) + "/report"` a `/` was missing

**Comment by uzaxirr** [comment]
Now the problem is if u hit https://www.filescan.io/api/scan/619f8affb71ceed41530067b/report for report u get a nice detailed response but in my case, I'm getting this 
```
{
  "flowId": "619f8affb71ceed41530067b",
  "reports": {},
  "priority": "least",
  "pollPause": 12,
  "allFinished": false,
  "reportsAmount": 0
}
```
`allFinished` should be true

Can u please figure out what went wrong

File: api_app/analyzers_manager/file_analyzers/filescan.py (hunk 2 above)
Comment (uzair-ali10) [reply api_app/analyzers_manager/file_analyzers/filescan.py]: Yes U are correct, fixed it now
it should be `+ "/report"` a `/` was missing

**Comment by sp35** [comment]
@uzair-ali10 This is why we need to poll for the results, the scan takes time.

**Comment by sp35** [comment]
I just checked https://www.filescan.io/api/scan/619f8affb71ceed41530067b/report

It's over now and the results are available.

**Comment by uzaxirr** [comment]
> w and the results are available.

@sp35 
Ohh now I understand the reason behind it, Can u also give me a reference to that polling function or how to do polling. So that i can go through it understand how to do it and implement the same

**Comment by uzaxirr** [comment]
@sp35 
From what I've understood by reading code for cuckoo_scan is i need to repeatedly send requests to https://www.filescan.io/api/scan/619f8affb71ceed41530067b/report in some time intervals until `allFinished` is `true` and then display the results
Correct?

**Pull request closed without merging**